pkg/controller/discovery/container/collection.go

```diff
@@ -131,8 +131,6 @@ func (r *SimpleReconciler) Reconcile(collection Collection) (err error) {
 			dpn.discovered = m
 		}
 	}
-	model.Mutex.RLock()
-	defer model.Mutex.RUnlock()
```

Comment:
Is the mutex not necessary at all here?

Reply:
The intent was to acquire the mutex for the duration of transaction.  The individual Insert(), Update() and Delete() will also call Lock() and because it is not reentrant will cause a panic.  It would be better to hold the lock for the duration of the transaction.  I'll investigate options.